src/Intersection.hs

```diff
@@ -0,0 +1,18 @@
+module Intersection where
+
+import Data.Ord
+
+import DataTypes
+
+data Intersection = Intersection {
+  intersectedObject :: Object,
+  distance :: Double
+} deriving Eq
+
+instance Ord Intersection where
+  compare x y
+    | distance x < distance y = LT
+    | distance x > distance y = GT
+    | otherwise = EQ
+  x <= y = distance x <= distance y
```

Comment:
yep, compare and `<=` are the minimum required for the Ord type class
https://hackage.haskell.org/package/base-4.8.2.0/docs/Data-Ord.html